### DISCUSSION TITLE
Migration from CRA - Uncaught ReferenceError: _Lock__WEBPACK_IMPORTED_MODULE_1__ is not defined

### DISCUSSION THREAD

**Discussion by maxim-semikov** [web-infra-dev/rsbuild#1633] [Q&A] [answered]
Hello!
I want to try migrate from CRA. So I decided create new project and try to do it.
I used UUI lib: https://github.com/epam/UUI
For starting I used template: `yarn create react-app my-app --template @epam/uui`
Than I used this guide: https://rsbuild.dev/guide/migration/cra

There isn't any warnings or errors in console for `dev` mode
But for `build` and than `preview` I got an error:
<img width="527" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/10222003/7abae5a7-8959-4ec9-b604-0d82504ecc15">
<img width="452" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/10222003/c871e930-dcfa-4765-ad06-034d409faa87">

this is my rsbuild.config.ts
```js
export default defineConfig({
  plugins: [pluginReact(), pluginSvgr()],
  html: {
    template: "./public/index.html",
  },
  server: {
    port: 3000,
  },
  source: {
    define: publicVars,
  },
  performance: {
    chunkSplit: {
      strategy: "split-by-module",
    },
  },
});

```

maybe someone know that I did wrong?

**Comment by xc2** [comment, marked as answer]
it might be a treeshaking bug.

pls try out the new treeshaking implementation with

```js
// rsbuild.config.ts
{
  tools: {
    rspack: {
      experiments: {
        rspackFuture: { newTreeshaking: true },
      },
    },
  },
}
```

or the workaround:

```js
import {FocusLockUI} from "react-focus-lock";
console.log(FocusLockUI);
```

reference: https://discord.com/channels/977448667919286283/1209798834767667230

**Reply by maxim-semikov** [reply]
Thanks! 
It works with new treeshaking

**Reply by chenjiahan** [reply]
We will enable the new treeshaking by default in Rspack / Rsbuild soon. 😄
